cpp/src/io/json/experimental/read_json.cpp

```diff
@@ -15,17 +15,48 @@
  */
 
 #include "read_json.hpp"
+#include <io/json/nested_json.hpp>
 
 #include <cudf/utilities/error.hpp>
+#include <io/comp/io_uncomp.hpp>
+
+#include <numeric>
 
 namespace cudf::io::detail::json::experimental {
 
+std::vector<uint8_t> ingest_raw_input(host_span<std::unique_ptr<datasource>> sources,
```

Comment:
I don't think we can, without an additional copy. Decompression returns vector<uint8_t> (stand-in for `byte`) since the data might be binary, not text as is the case with JSON.

Reply:
Okay. Maybe a refactoring towards std::byte can be done sometime.

Reply:
Yeah, I have that on the radar. The problem is that it leads to bigger mess unless it's done consistently (and that's a lot of code to change!).